### PR TITLE
Added SMT, CPU workload support for different components(Ex: DLPAR)

### PR DESCRIPTION
**Pull request by SamirMulani**
1. Added a new test case to run SMT workload as a daemon process.
2. Added a new test case to run CPU workload as a daemon process.

### CODE REVIEW

**Comment by SamirMulani** [comment]
Run Logs,
```
# avocado run smt.py:smt.test_smt_start
JOB ID     : bd03c448069d803f587109bfe4d9c80db4e5f476
JOB LOG    : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.11-bd03c44/job.log
 (1/1) smt.py:smt.test_smt_start: STARTED
 (1/1) smt.py:smt.test_smt_start: PASS (0.41 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.11-bd03c44/results.html
JOB TIME   : 8.38 s
# avocado run smt.py:smt.test_smt_stop
JOB ID     : d4f8ca2cf8648fbb955cd625fb354558aeaeb0b6
JOB LOG    : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.11-d4f8ca2/job.log
 (1/1) smt.py:smt.test_smt_stop: STARTED
 (1/1) smt.py:smt.test_smt_stop: PASS (0.53 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.11-d4f8ca2/results.html
JOB TIME   : 11.59 s


# avocado run cpu_fold.py:cpu_workload.test_cpu_start
JOB ID     : c4143160c97eaf2e7377f9e0a89a365079f1ce70
JOB LOG    : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.12-c414316/job.log
 (1/1) cpu_fold.py:cpu_workload.test_cpu_start: STARTED
 (1/1) cpu_fold.py:cpu_workload.test_cpu_start: PASS (0.42 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.12-c414316/results.html
JOB TIME   : 8.59 s
# avocado run cpu_fold.py:cpu_workload.test_cpu_stop
JOB ID     : 9c8ea6b68e098d41a11fdc706727a8ebba7b4d56
JOB LOG    : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.13-9c8ea6b/job.log
 (1/1) cpu_fold.py:cpu_workload.test_cpu_stop: STARTED
 (1/1) cpu_fold.py:cpu_workload.test_cpu_stop: PASS (0.54 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /Workspace/avocado-fvt-wrapper/results/job-2023-04-28T15.13-9c8ea6b/results.html
JOB TIME   : 8.62 s
```

File: workload/cpu_fold.py.data/cpu.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ppc64_cpu --cores-on
+ppc64_cpu --info
+sleep 2
+
+ppc64_cpu --online-cores=4,5,6,7
+ppc64_cpu --cores-on
+ppc64_cpu --info
+sleep 2
+
+ppc64_cpu --offline-cores=10,11,13,15
```
[the remark this answers is not in the document]
Comment (abdhaleegit) [reply workload/cpu_fold.py.data/cpu.sh:34]: @Geet2 @SamirMulani the .sh itself can get you the number of procs by using this command

`nproc`

```
[abdul@abdul ]$ nproc
16
```

[the remark this answers is not in the document]
Comment (SamirMulani) [reply workload/cpu_fold.py.data/cpu.sh:34]: Yes @abdhaleegit I have implemented it with the help of **nproc** Linux command.

File: workload/cpu_fold.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        sm = SoftwareManager()
+        for pkg in ['util-linux', 'powerpc-utils', 'numactl']:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.cancel("%s is required to continue..." % pkg)
+
+    def test_cpu_start(self):
+        """
+        Start the CPU Workload
+        """
+        self.cwd = os.getcwd()
```
Comment (PraveenPenguin) [inline workload/cpu_fold.py:42]: can we eliminate using extra variables?

File: workload/cpu_fold.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        self.smt_workload = self.cwd + "/" + \
```
Comment (PraveenPenguin) [inline workload/cpu_fold.py:43]: please have variable scope as local , even rename it logical name as this for cpu_folding

File: workload/cpu_fold.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            "cpu_fold.py.data/cpu.sh &> /tmp/cpu.log &"
+        self.cmd = "bash " + self.smt_workload
```
Comment (PraveenPenguin) [inline workload/cpu_fold.py:45]: can we eliminate this variable as well

File: workload/cpu_fold.py.data/cpu.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# An online and offline core workload script to stress the 
+# cpu's on- and offline core functionality using the number 
+# of cores available on the system.
+
+set -x
```
Comment (PraveenPenguin) [inline workload/cpu_fold.py.data/cpu.sh:6]: why set -x , for debug? can we remove this

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]
@SamirMulani please address the style comment (same will be applicable to smt workload as well ) and add a check if the test is not running on PowerPC it should be canceled

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**